src_cam/physpkg.F90

```diff
@@ -1900,17 +1953,54 @@ subroutine tphysac (ztodt,   cam_in,  &
                   fh2o, surfric, obklen, flx_heat, cmfmc, dlf, det_s, det_ice, net_flx)
           end if
 
-          call check_energy_cam_chng(state, tend, "microp_tend", nstep, ztodt, &
-               zero, prec_str(:ncol)/cld_macmic_num_steps, &
-               snow_str(:ncol)/cld_macmic_num_steps, zero)
-
+          if (compute_enthalpy_flux) then
+             write(physparname,"(i3)") macmic_it
+             physparname="microp_tend "//trim(physparname)
+             call check_energy_cam_chng(state, tend, physparname, nstep, ztodt, &
+                  zero, prec_str(:ncol)/cld_macmic_num_steps, &
+                  snow_str(:ncol)/cld_macmic_num_steps, zero)
+          else
+             call check_energy_cam_chng(state, tend, "microp_tend", nstep, ztodt, &
+                  zero, prec_str(:ncol)/cld_macmic_num_steps, &
+                  snow_str(:ncol)/cld_macmic_num_steps, zero)
+          end if
           call t_stopf('microp_tend')
 
           prec_sed_macmic(:ncol) = prec_sed_macmic(:ncol) + prec_sed(:ncol)
           snow_sed_macmic(:ncol) = snow_sed_macmic(:ncol) + snow_sed(:ncol)
           prec_pcw_macmic(:ncol) = prec_pcw_macmic(:ncol) + prec_pcw(:ncol)
           snow_pcw_macmic(:ncol) = snow_pcw_macmic(:ncol) + snow_pcw(:ncol)
 
+          if (compute_enthalpy_flux) then
+             if(macmic_it.eq.1) then
+                qcsedten_idx = pbuf_get_index('QCSEDTEN'     , errcode=i)
+                qrsedten_idx = pbuf_get_index('QRSEDTEN'     , errcode=i)
+                qisedten_idx = pbuf_get_index('QISEDTEN'     , errcode=i)
+                qssedten_idx = pbuf_get_index('QSSEDTEN'     , errcode=i)
+                qgsedten_idx = pbuf_get_index('QGSEDTEN'     , errcode=i)
+             endif
```

Comment:
This is a bit awkward. I suggest moving the `pbuf_get_index` calls to `phys_init` (without the `macmic` logic) to avoid calling these routines every timestep.